src/autotype/AutoType.cpp

```diff
@@ -423,6 +425,11 @@ void AutoType::performGlobalAutoType(const QList<QSharedPointer<Database>>& dbLi
         return;
     }
 
+    // Invalidate last match if it's old enough
+    if (m_lastMatch.first && (QDateTime::currentMSecsSinceEpoch() - m_lastMatchTime) > 30000) {
```

Comment:
Put the 30000 in a const int within anonymous namespace or a define. Also use Clock:: interface instead of QDateTime directly.

Reply:
Done, converted to using the seconds since we don't need more accuracy.